### PR TITLE
Docs: Add Private Data Source Connect docs

### PR DESCRIPTION
**Pull request by mjseaman**
Add documentation for private data source connect, available in Cloud Pro and Advanced in closed preview

**What is this feature?**

Add documentation for private data source connect, available in Cloud Pro and Advanced in closed preview. These docs should be public, but not indexed or searchable. 

**Who is this feature for?**

Closed preview customers, who will all be given a link to the docs.

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer**:

I think this is a reasonable place to put these docs, but this feature is only available in Grafana Cloud, so I wonder if this doc should go into Grafana Cloud documentation...

### CODE REVIEW

File: docs/sources/setup-grafana/configure-security/configure-private-datasource-connect.md [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+
+        Additionally:
+
+         - ${PDC_GATEWAY}: The URL of the private data source connect in Grafana Cloud. The Grafana team will give you this URL. It will follow the format `grafana-private-datasource-connect-&lt;cluster>.grafana.net`
+         - ${SLUG}: The name of the stack you would like to connect to your data source. For example, the stack “test.grafana.net” has the slug “test.” 
+    - **Option 2:** Using the [pdc-agent](https://github.com/grafana/pdc-agent) docker [image](https://hub.docker.com/r/grafana/pdc-agent/tags)
+
+
+        ```
+        docker run --rm --name pdc-agent -v $(pwd):/etc/keys grafana/pdc-agent:latest -i /etc/keys/${SLUG} ${SLUG}@host.docker.internal -p 2222 -o BatchMode=yes -o UserKnownHostsFile=/etc/keys/known_hosts -o CertificateFile=/etc/keys/${SLUG}-cert.pub -R 0 -v
```
Comment (chaudyg) [inline docs/sources/setup-grafana/configure-security/configure-private-datasource-connect.md:123]:
```suggestion
        docker run --rm --name pdc-agent -v $(pwd):/etc/keys grafana/pdc-agent:latest -i /etc/keys/${SLUG} ${SLUG}@${PDC_GATEWAY} -p 22 -o BatchMode=yes -o UserKnownHostsFile=/etc/keys/known_hosts -o CertificateFile=/etc/keys/${SLUG}-cert.pub -R 0 -v
```

File: docs/sources/setup-grafana/configure-security/configure-private-datasource-connect.md [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+
+
+
+        The flags used on this are a combination of:
+
+        - –-rm: Remove the docker container when it exits.
+        - --name pdc-agent: This names the docker process pdc-agent
+        - -v $(pwd):/etc/keys: Copies the working directory into the /etc/keys directory in the Docker container.
+        - -i /etc/keys/${SLUG}: The private key
+        - -p 2222: The docker container port
```
Comment (chaudyg) [inline docs/sources/setup-grafana/configure-security/configure-private-datasource-connect.md:134]:
```suggestion
        - -p 22: The port to connect to
```

**Review by chri2547 — CHANGES_REQUESTED** [review]
One small change to image. Thank you!

**Review by chri2547 — APPROVED** [review]
Great stuff! Thank you!

File: docs/sources/setup-grafana/configure-security/configure-private-datasource-connect/index.md [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+     - -o [CertificateFile](https://man.openbsd.org/ssh_config.5#CertificateFile): Your public certificate
+     - -R 0: Runs ssh with remote port forwarding, which enables it to act as a socks server
+     - -vv (optional): Sets the verbosity to debug2, so hostnames can be seen. It can be set to -v, if this is not desired.
+     - -o [PermitRemoteOpen](https://man.openbsd.org/ssh_config.5#PermitRemoteOpen) (optional): This can be specified to restrict the destinations reachable by Grafana Cloud over this connection.
+
+     Refer to the [OpenSSH documentation](https://linux.die.net/man/1/ssh) for a complete list of available ssh command flags.
+
+     Additionally:
+
+     - ${PDC_GATEWAY}: The URL of the private data source connect in Grafana Cloud. The Grafana team will give you this URL. The URL follows the format `grafana-private-datasource-connect-&lt;cluster>.grafana.net`
```
Comment (chaudyg) [inline docs/sources/setup-grafana/configure-security/configure-private-datasource-connect/index.md:118]:
```suggestion
     - ${PDC_GATEWAY}: The URL of the private data source connect in Grafana Cloud. The Grafana team will give you this URL. The URL follows the format `private-datasource-connect-<cluster>.grafana.net`
```

**Pull request merged**